core/src/main/kotlin/org/evomaster/core/output/naming/ActionTestCaseNamingStrategy.kt

```diff
@@ -0,0 +1,48 @@
+package org.evomaster.core.output.naming
+
+import com.webfuzzing.commons.faults.FaultCategory
+import org.evomaster.core.output.TestWriterUtils
+import org.evomaster.core.problem.enterprise.DetectedFaultUtils
+import org.evomaster.core.problem.httpws.HttpWsCallResult
+import org.evomaster.core.problem.rest.RestCallAction
+import org.evomaster.core.search.EvaluatedIndividual
+import org.evomaster.core.search.Solution
+import org.evomaster.core.search.action.EvaluatedAction
+
+open class ActionTestCaseNamingStrategy(
+    solution: Solution<*>
+) : NumberedTestCaseNamingStrategy(solution)  {
+
+
+    override fun expandName(individual: EvaluatedIndividual<*>): String {
+        var evaluatedAction = individual.evaluatedMainActions().last()
+        var action = evaluatedAction.action as RestCallAction
+
+        return "_${action.verb}_on_${getPath(action.path.nameQualifier)}_${addResult(individual)}"
+    }
+
+    private fun getPath(nameQualifier: String): String {
+        if (nameQualifier == "/") {
+            return "root"
+        }
+        return TestWriterUtils.safeVariableName(nameQualifier)
+    }
+
+    private fun addResult(individual: EvaluatedIndividual<*>): String {
+        val detectedFaults = DetectedFaultUtils.getDetectedFaultCategories(individual)
+        if (detectedFaults.isNotEmpty()) {
+            return fault(detectedFaults)
+        }
+        return statusCode(individual.evaluatedMainActions().last())
+    }
+
+    private fun fault(faults: Set<FaultCategory>): String {
```

Comment:
if it is only 1, then yes, using `testCaseLabel` makes sense.
but, if it is more than 1 in the test (and not just last call), then maybe using sorted list of unique `code` could be better, eg something like `Faults_100_101_400_500_801`